apps/mobile/src/kernel/navigation/AppNavigator.tsx

```diff
@@ -48,127 +50,135 @@ export const AppNavigator = () => {
-              getComponent={() => ChooseBiometricLoginScreen}
-            />
-          )}
+        {/* Authenticated */}
 
-          {afterLoginAction === 'dark-theme-announcement' && (
-            <Stack.Screen //
-              name="dark-theme-announcement"
-              options={{headerShown: false}}
-              getComponent={() => DarkThemeAnnouncementScreen}
-            />
-          )}
+        {isLoggedIn && (
+          <Stack.Group>
+            {afterLoginAction === 'choose-biometric-login' && (
+              <Stack.Screen //
+                name="choose-biometric-login"
+                options={{headerShown: false}}
+                getComponent={() => ChooseBiometricLoginScreen}
+              />
+            )}
 
-          {afterLoginAction === 'setup-wallet' && (
-            <Stack.Screen //
-              name="setup-wallet"
-              options={{headerShown: false}}
-              component={SetupWalletNavigator}
-            />
-          )}
+            {afterLoginAction === 'dark-theme-announcement' && (
+              <Stack.Screen //
+                name="dark-theme-announcement"
+                options={{headerShown: false}}
+                getComponent={() => DarkThemeAnnouncementScreen}
+              />
+            )}
 
-          {afterLoginAction === 'manage-wallets' && (
-            <Stack.Screen
-              name="manage-wallets"
-              getComponent={() => WalletNavigator}
-            />
-          )}
-        </Stack.Group>
-      )}
+            {afterLoginAction === 'setup-wallet' && (
+              <Stack.Screen //
+                name="setup-wallet"
+                options={{headerShown: false}}
+                component={SetupWalletNavigator}
+              />
+            )}
 
-      {/* Modal */}
+            {afterLoginAction === 'manage-wallets' && (
+              <Stack.Screen
+                name="manage-wallets"
+                getComponent={() => WalletNavigator}
+              />
+            )}
+          </Stack.Group>
+        )}
 
-      <Stack.Group
-        screenOptions={{
-          gestureEnabled: false,
-          presentation: 'transparentModal',
-          ...(Platform.OS === 'android' && {
-            ...TransitionPresets.DefaultTransition,
-          }), // overriding general navigation settings
-          cardStyle: a.bg_transparent, // this is needed for the modal to be transparent
-        }}
-      >
-        <Stack.Screen
-          name="modal"
-          getComponent={() => Modal}
-          options={{
+        {/* Modal */}
+
+        <Stack.Group
+          screenOptions={{
             gestureEnabled: false,
+            presentation: 'transparentModal',
+            ...(Platform.OS === 'android' && {
+              ...TransitionPresets.DefaultTransition,
+            }), // overriding general navigation settings
+            cardStyle: a.bg_transparent, // this is needed for the modal to be transparent
           }}
```

Comment:
i think this modal should be removed. legacy

Reply:
Sure, I'll delete it, I was just moving things in the file, didn't really add it :D